62.SchusslerTest/shader.frag

```diff
@@ -0,0 +1,127 @@
+#version 430 core
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+
+layout (location = 0) in vec3 Normal;
+layout (location = 1) in vec3 Pos;
+layout (location = 2) flat in vec3 LightPos;
+layout (location = 3) flat in vec3 CamPos;
+layout (location = 4) flat in vec3 GNormal;
+layout (location = 5) flat in float Alpha;
+layout (location = 6) flat in float AlphaY;
+
+layout (location = 0) out vec4 outColor;
+
+layout (push_constant) uniform PC {
+    layout (offset = 64) vec3 campos;
+    layout (offset = 80) uint testNum;
+} pc;
+
+#define TEST_GGX 1
+#define TEST_BECKMANN 2
+#define TEST_PHONG 3
+#define TEST_AS 4
+#define TEST_OREN_NAYAR 5
+#define TEST_LAMBERT 6
+
+#include <nbl/builtin/glsl/bxdf/brdf/specular/ggx.glsl>
+#include <nbl/builtin/glsl/bxdf/brdf/specular/beckmann.glsl>
+#include <nbl/builtin/glsl/bxdf/brdf/specular/blinn_phong.glsl>
+#include <nbl/builtin/glsl/bxdf/brdf/diffuse/oren_nayar.glsl>
+#include <nbl/builtin/glsl/bxdf/brdf/diffuse/lambert.glsl>
+
+float pdot(vec3 a, vec3 b) {
+    return max(0.0, dot(a,b));
+}
+
+float ap(vec3 wi, vec3 wp, vec3 wg) {
+    return pdot(wi, wp)/pdot(wp, wg);
+}
+
+float at(vec3 wi, vec3 wt, vec3 wp, vec3 wg) {
+    return pdot(wi, wt) * sqrt(1-(pdot(wp,wg)*pdot(wp,wg))) / pdot(wp,wg);
+}
```

Comment:
you don't use these functions anywhere but for `lambda`, so can remove and have optimized variants

Reply:
also use `1.0` or `1.f` instead of 1, cause GLSL is a stickler for types